server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -126,12 +122,58 @@ public SetMessagesResponse process(SetMessagesRequest request, MailboxSession ma
+                .map(box -> box.getMailboxId().serialize())
+                .map(id -> messageWithId.getMailboxIds().contains(id))
+                .orElse(false);
+        return validMessagesTester.test(messageWithId) && isMessageSetInOutbox;
+    }
+
+    private Predicate<CreationMessage> getNotImplementedRequestTester(MailboxSession mailboxSession) {
+        Optional<Id> draftsId = systemMailboxesProvider.getStreamOfMailboxesFromRole(Role.DRAFTS, mailboxSession)
+                .map(Mailbox::getMailboxId)
+                .findFirst();
+        return e -> draftsId
+                .map(MailboxId::serialize)
+                .map(id -> e.getMailboxIds().contains(id))
+                .orElse(false);
+    }
+
+    private Map<CreationMessageId, SetError> handleCreationErrors(SetMessagesRequest request,
+                                                                  Predicate<CreationMessage> invalidMessagesTester,
+                                                                  Predicate<CreationMessage> notImplementedPredicate) {
+
+        Set<Map.Entry<CreationMessageId, CreationMessage>> creationRequestEntries = request.getCreate().entrySet();
+        Map<CreationMessageId, SetError> mapOfNotImplEntries = getMapOfNotImplErrors(notImplementedPredicate, creationRequestEntries);
+        List<AbstractMap.SimpleEntry<CreationMessageId, SetError>> invalidArgumentsErrors = getInvalidArgumentsErrors(creationRequestEntries, invalidMessagesTester, mapOfNotImplEntries);
+        // Merge maps of SetErrors
+        return Stream.concat(mapOfNotImplEntries.entrySet().stream(), invalidArgumentsErrors.stream())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private Map<CreationMessageId, SetError> getMapOfNotImplErrors(Predicate<CreationMessage> notImplementedPredicate,
+                                                                   Set<Map.Entry<CreationMessageId, CreationMessage>> creationRequestEntries) {
+
+        SetError notImplementedError = SetError.builder().type("error").description("Not yet implemented").build();
+
+        return creationRequestEntries.stream()
+                .filter(e -> notImplementedPredicate.test(e.getValue()))
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> notImplementedError));
+    }
+
+    private List<AbstractMap.SimpleEntry<CreationMessageId, SetError>> getInvalidArgumentsErrors(
+            Set<Map.Entry<CreationMessageId, CreationMessage>> creationRequestEntries,
+            Predicate<CreationMessage> invalidMessagesTester, Map<CreationMessageId, SetError> mapOfNotImplEntries) {
+
+        // Deal with invalid arguments but only for entries that have not been flagged yet (ie not impl...)
+        return creationRequestEntries.stream()
+                .filter(e -> !mapOfNotImplEntries.containsKey(e.getKey())) // exclude requests that have already been flagged with "not implemented"
```

Comment:
I really don't like this pattern, but ok for this time, should refactor next time.